lib/internal/utils.dart

```diff
@@ -16,7 +22,44 @@ class Utils {
       ),
     );
   }
+
   static bool statusFromResponse(Response response){
     return json.decode(response.body)["status"];
   }
+
+  static Map<String, dynamic> toSyncMap(Note note){
+    var originalMap = note.toJson();
+    Map<String, dynamic> newMap = Map();
+    originalMap.forEach((key, value) {
+      var newValue = value;
+      var newKey = ReCase(key).snakeCase;
+      switch(key){
+        case "styleJson":{
+          var style = value as ContentStyle;
+          newValue = json.encode(style.data);
+          break;
+        }
+        case "images":{
+          var images = value as ImageList;
+          newValue = json.encode(images.images);
+          break;
+        }
+        case "listContent":{
+          var listContent = value as ListContent;
+          newValue = json.encode(listContent.content);
+          break;
+        }
+        case "reminders":{
+          var reminders = value as ReminderList;
+          newValue = json.encode(reminders.reminders);
+          break;
+        }
+      }
+      if(key == "id"){
+        newKey = "note_id";
+      }
+      newMap.putIfAbsent(newKey, () => newValue);
+    });
+    return newMap;
```

Comment:
if you always need the encoded json just put it on here, maybe with a bool param to omit encoding

Reply:
I'll prolly just make a small function for it like toSyncJson(Note note)